### ISSUE TITLE
TaaT algorithms don't work with new parallel `evaluate_queries`

### ISSUE THREAD

**Issue by JMMackenzie** [pisa-engine/pisa#201]
**Describe the bug**
When you use any of the TaaT algorithms with `evaluate_queries`, it throws a `bad_alloc` - I think this has something to do with the accumulators being init once and used by all (perhaps). Will happen even if we set `--threads 1`.

**To Reproduce**
1. Run `evaluate_queries` with any of the TaaT algos.

**Error message**
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
```

**Expected behavior**
Program should run and dump trec results as normal.

**Environment info**
Operating System: Ubuntu
Compiler name: GCC (g++)
Compiler version: 7.3.0

**Comment by amallia** [comment]
The issue seems to be related to the fact that multiple threads run the same lambda function:
https://github.com/pisa-engine/pisa/blob/967bd65fc13077da9df2ca90b6c5e3ee0744d2ef/src/evaluate_queries.cpp#L134-L137

**Comment by JMMackenzie** [comment]
This seems to be solved for me now (via #234) - Could you double check? We can close this one off.

**Comment by amallia** [comment]
Solved!
